ambry-api/src/main/java/com.github.ambry/config/ConnectionPoolConfig.java

```diff
@@ -60,6 +60,14 @@
   @Default("2")
   public final int connectionPoolMaxConnectionsPerPortSSL;
 
+  /**
+   * The maximum consecutive connect failures allowed before we clean up the available pool
+   */
+  @Config("connectionpool.max.connect.failures.to.clean.up.available.pool")
```

Comment:
Yes. In one of our tests, we noticed that frontends were not marking server nodes as up after 5 mins even if the server is up(server was unreachable initially due to deployment). After investigation, we figured out the reason. BlockingChannelConnectionPool had N no of stale connections when the destination host went down. When the frontend started to issue request after some time to check if the destination is up, it was given a stale connection from the pool. So, it would fail eventually and hence the node was left as down. This continues and after a long time( when eventually the pool is exhausted for new checkouts), we start fresh ones and the frontend marks the node as up. Hope you can get the reason behind this fix.